contracts/contracts/DC.sol

```diff
@@ -6,135 +6,71 @@ import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/security/Pausable.sol";
 
 import "./IRegistrarController.sol";
+import "./INameWrapper.sol";
 import "./IBaseRegistrar.sol";
 
 /**
     @title A domain manager contract for .country (DC -  Dot Country)
     @author John Whitton (github.com/johnwhitton), reviewed and revised by Aaron Li (github.com/polymorpher)
     @notice This contract allows the rental of domains under .country (”DC”)
     it integrates with the ENSRegistrarController and the ENS system as a whole for persisting of domain registrations.
-    It is responsible for holding the revenue from these registrations for the web2 portion of the registration process,
-    with the web3 registration revenue being held by the RegistrarController contract.
-    An example would be as follows Alice registers alice.com and calls the register function with an amount of 10,000 ONE.
-    5000 ONE would be held by the DC contract and the remaining 5000 funds would be sent to the RegistrarController using 
-    the register function.
+    The calling services are responsible for holding the revenue from these registrations for the web2 portion of the 
+    registration process, with the web3 registration revenue being held by the RegistrarController contract.
+    An example would be as follows Alice registers alice.com this calls the service register function with an amount of 10,000 ONE.
+    5000 ONE would be held by the service contract and the remaining 5000 funds would be sent to the RegistrarController 
+    via this contract using the register function.
 
  */
 contract DC is Pausable, Ownable {
-    uint256 public gracePeriod;
-    uint256 public baseRentalPrice;
-    address public revenueAccount;
     IRegistrarController public registrarController;
+    INameWrapper public nameWrapper;
     IBaseRegistrar public baseRegistrar;
-    uint256 public duration;
     address public resolver;
     bool public reverseRecord;
     uint32 public fuses;
     uint64 public wrapperExpiry;
     bool public initialized;
 
     struct InitConfiguration {
-        uint256 baseRentalPrice;
-        uint256 duration;
-        uint256 gracePeriod;
 
-        // 32-bytes block
-        address revenueAccount;
+        // 12-bytes block
```

Comment:
these fields should be aligned to 32-byte blocks or the alignment won't be meaningful at all